### ISSUE TITLE
ICE for enums with int initializers

### ISSUE THREAD

**Issue by jmr** [cc65/cc65#1245]
```
enum E {
    X = 1000,
} e = 3;
```

Produces:
```
cc65: Check failed: IsClassStruct (T) || IsTypeEnum (T), file 'cc65/datatype.c', line 1208
```

@acqn The (immediate) problem seems to be here:
https://github.com/cc65/cc65/blame/master/src/cc65/typecmp.c#L203

```
        if ((IsTypeEnum (lhs) || IsTypeEnum (rhs))) {

            /* Compare the tag types */
            Sym1 = GetESUSymEntry (lhs);
            Sym2 = GetESUSymEntry (rhs);
```

`rhs` is not an `enum`, but `GetESUSymEntry` is called on it.

**Comment by jmr** [comment]
Blocking #1244.

**Comment by acqn** [comment]
Yep. The precondition of `GetESUSymEntry` was changed to be narrower. Will fix it.

**Comment by acqn** [comment]
@jmr Fixed in PR #1246.

**Comment by jmr** [comment]
Thanks!
